### DISCUSSION TITLE
Help with inlabru data form and general

### DISCUSSION THREAD

**Discussion by rogerio-bio** [inlabru-org/inlabru#190] [Q&A]
Hey guys,

So, reading the inlabru article the package seems to have what I need, but the tutorials are too confusing for me.

I need to create a prey density map with two covariates: distance from edge and distance from rivers. I have these two variables in raster form. I have the Long/Lat of the traps and the number of occurrences of nine species in each of the traps (which ranges from 0 - 233).

The question is: Do covariates need to be inside the table along with my data?

Can I use long/lat in decimals or do they need to be in UTM?

Any help or guide I appreciate it

**Comment by finnlindgren** [comment]
Hi, the details of the answer depends on what observation model you want to implement; if it's just a Poisson count model with a count for each trap, conditionally independent given the spatial field, there are more options than for some other models. E.g., you could choose to precompute the covariate values at the trap locations and include in the data itself.  Or you can use `eval_spatial(rastercovariate, .data.)` as component input, to evaluate it "on the fly" based on `sp` or `sf` point data.

In recent inlabru versions you should be able to use any valid CRS coordinate system, so long/lat is ok as long as your CRS information specifies that, and same thing with UTM.
